src/BunqLib.php

```diff
@@ -239,12 +245,30 @@ public function makePayment(
         // Create a new payment and retrieve it's id.
         return Payment::create(
             new Amount($amount, self::CURRENCY_TYPE_EUR),
-            new Pointer(self::POINTER_TYPE_EMAIL, $recipient),
+            $this->determinePointerFromRecipient($recipient),
             $description,
             $monetaryAccount->getId()
         )->getValue();
     }
 
+    /**
+     * @param string $recipient
+     *
+     * @return Pointer
+     * @throws BunqException
+     */
+    public function determinePointerFromRecipient(string $recipient): Pointer
+    {
+        if (filter_var($recipient, FILTER_VALIDATE_EMAIL)) {
+            $pointerType = self::POINTER_TYPE_EMAIL;
+        } elseif (preg_match(self::REGEX_E164_PHONE, $recipient)) {
+            $pointerType = self::POINTER_TYPE_PHONE_NUMBER;
+        } else {
+            $pointerType = self::POINTER_TYPE_IBAN;
+        }
+
+        return new Pointer($pointerType, $recipient);
```

Comment:
If pointer is type IBAN, then you must also pass `name` value to the constructor.  If not backend will return missing name :P So you should restructure this :D.